pkg/autoscaling/calculation.go

```diff
@@ -186,8 +194,53 @@ func getCPUByResourceType(strategy *Strategy, resourceType string) uint64 {
 }
 
 // TODO: get the scaled groups
-func getScaledGroupsByComponent(component ComponentType) []*Plan {
-	return []*Plan{}
+func getScaledGroupsByComponent(rc *cluster.RaftCluster, component ComponentType, healthyInstances []instance) []*Plan {
+	switch component {
+	case TiKV:
+		return getScaledTiKVGroups(rc, healthyInstances)
+	case TiDB:
+		// TODO: support search TiDB Group
+		return []*Plan{}
+	default:
+		return nil
+	}
+}
+
+func getScaledTiKVGroups(informer core.StoreSetInformer, healthyInstances []instance) []*Plan {
+	var plans []*Plan
+	planMap := make(map[string]map[uint64]struct{}, len(healthyInstances))
+	for _, instance := range healthyInstances {
+		store := informer.GetStore(instance.id)
+		if store == nil {
+			log.Warn("inconsistency between healthInstances and StoreSetInformer, exit auto-scaling calculation",
+				zap.Uint64("StoreID", instance.id))
```

Comment:
```suggestion
				zap.Uint64("store-id", instance.id))
```
BTW, how about making the log more readable by hiding these variable names?